app/api/recipes/[id]/route.ts

```diff
@@ -0,0 +1,171 @@
+// app/api/recipes/[id]/route.ts
+import { NextResponse } from "next/server";
+import {
+  getRecipeById,
+  updateRecipe,
+  deleteRecipe,
+} from "@/controller/recipeController";
+import { uuidSchema } from "@/validation/uuidValidation";
+import { recipeUpdateSchema } from "@/validation/recipeValidation";
+
+/**
+ * GET /api/recipes/{id} - Retrieves a recipe.
+ * @summary Get a specific recipe by ID.
+ * @param req - The request object.
+ * @param params - An object with the recipe id.
+ * @returns JSON with recipe data or an error message.
+ */
+export async function GET(
+  req: Request,
+  { params }: { params: Promise<{ id: string }> }
+): Promise<NextResponse> {
+  let id: string;
+
+  // Get ID from path parameters.
+  try {
+    const resolvedParams = await params;
+    id = resolvedParams.id;
+    if (!id) throw new Error();
+  } catch {
+    return NextResponse.json("Not Found", { status: 404 });
+  }
+
+  // Validate that the recipe ID is a UUID.
+  let { error } = await uuidSchema.validate({ uuid: id });
+  if (error) {
+    return NextResponse.json("Bad Request - Invalid Recipe ID", {
+      status: 400,
+    });
```

Comment:
Don't break up a line to stay within the ~80 character confine if it hinders readability. It's just a rule of thumb.


```suggestion
    return NextResponse.json("Bad Request - Invalid Recipe ID", {status: 400});
```